fineract-provider/build.gradle

```diff
@@ -324,7 +324,6 @@ tasks.withType(JavaCompile) {
                 "BadImport",
                 "UnusedMethod",
                 "ModifiedButNotUsed",
-                "ArgumentSelectionDefectChecker",
```

Comment:
@percyashu I see on https://errorprone.info/bugpattern/ArgumentSelectionDefectChecker that by default this is just a Warning. Given that this seems to fix real bugs (which you are seem to be proposing to fix in this PR? I have not looked closely at each...), should we not configure this, and every other currently disabled check as you go along, to actually be an Error instead of just a Warning?